py-rse/git-cmdline.Rmd

```diff
@@ -10,6 +10,31 @@ These interfaces are actually wrappers around Git's original command-line interf
 which allows us to access all of Git's features.
 This lesson describes how the basic cycle of add, commit, push, and pull works at the command line.
 
+To illustrate with the Zipf's Law project, 
+we will start by putting our work up to this point in git.  
+Then, to illustrate the version control cycle, 
+we'll edit our, `plotcounts.py` script to produce a more useful plot.
```

Comment:
I believe that the starting script should look something like the below at this point. @DamienIrving Could you confirm?

```python
"""Plot word counts."""
import argparse
import pandas as pd


def main(args):
    df = pd.read_csv(args.infile, header=None, names=('word', 'word_frequency'))
    df['rank'] = df['word_frequency'].rank(ascending=False)
    df['inverse_rank'] = 1 / df['rank']
    ax = df.plot.scatter(x='word_frequency', y='inverse_rank',
                         figsize=[12, 6], grid=True)
    ax.figure.savefig(args.outfile)


if __name__ == '__main__':
    parser = argparse.ArgumentParser(description=__doc__)
    parser.add_argument('infile', type=argparse.FileType('r'), nargs='?',
                        default='-', help='Word count csv file name')
    parser.add_argument('--outfile', type=str, default='plotcounts.png',
                        help='Output image file name')
    parser.add_argument('--xlim', type=float, nargs=2, metavar=('XMIN', 'XMAX'),
                        default=None, help='X-axis limits')
    args = parser.parse_args()
    main(args)
```

Reply:
@joelostblom I guess you could tell I was working from a slightly different version...

What's the best way to handle this?  Should I just plug this in verbatim inline, or is there a better way to handle it (e.g. a link to somewhere permanent in our GitHub)?

Reply:
I say for now, just put this inline as is and if we want we can easily move it to a script and reference its path instead.

Maybe introduce with a better worded version of this: "There are slightly different approaches to solve the mission critical exercise at the end of last chapter. Since we will be adding to the resulting script from that exercise, we provide the full code here so that you can follow along with the incremental changes we make in this chapter.